advisories/a63162.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Preliminary Technical Advisory 63162
+summary: Invalid incremental backups under certain circumstances
+toc: true
+---
+
+Preliminary date: April 6, 2021
+
+**Preliminary, and subject to change**
+
+## Description
+
+Cockroach Labs has discovered a bug relating to incremental backups, which under certain circumstances may result in data loss when restoring from an incremental backup. This is a **preliminary notice** in the interest of transparency, and will change.
+
+If a backup coincides with an in-progress index creation (backfill), `RESTORE`, or `IMPORT`, it is possible that a subsequent incremental backup will not include all of the indexed, restored or imported data.
+
+Full backups are unaffected by this issue.
+
+This issue affects CockroachDB [versions](/docs/releases/) 19.1 and later.
+
+## Statement
+
+We are investigating and implementing fixes to the above-described issue. You should expect follow-up from Cockroach Labs in the near future.
+
+We have deliberately chosen to document this issue before the fixes are available. We believe the alternative -- of waiting for the fixes prior to communicating -- would be less responsible.
+
+The upcoming patch release will include fixes, detection and recovery tools, and is expected in a small number of weeks.
+
+To our knowledge, this issue has not been observed to affect any user. It was discovered via an internal investigation by Cockroach Labs engineers, and dates back to CockroachDB version 19.1, released in 2019.
+
+You can track progress on the following GitHub issues and pull requests:
+
+- [#63162](https://github.com/cockroachdb/cockroach/issues/63162)
+- [#62564](https://github.com/cockroachdb/cockroach/issues/62564) 
+- [#62565](https://github.com/cockroachdb/cockroach/issues/62565) 
+- [#62572](https://github.com/cockroachdb/cockroach/pull/62572) 
+
+You can track our releases, where fixed versions will appear:
+
+- [All Production Releases](../releases/index.html)
+
+You can track our technical advisories, including this one, here:
+
+- [Technical Advisories](index.html)
+
+Questions about this technical alert can be directed to our [support team](https://support.cockroachlabs.com/), or to your customer success contact at Cockroach Labs.
+
+## Mitigation
+
+We will have fixes, detection tools, and mitigation advice in the near future, as patch releases to several CockroachDB versions. If you wish to operate with an abundance of caution in the meantime:
+
+- Only consider full backups to be reliable. Consider incremental backups to be unreliable, if those incremental backups were taken during in-progress index backfills or imports.
+- If you redefine primary indexes, take a full backup before and after the operation; do not rely on incremental backups taken while doing this operation.
+- If you use incremental backups, refrain from doing imports until a fix is available.
```

Comment:
I think this could be more narrowly scoped: you can do keep doing inc backups, and you can trust them for all your online, production tables, but if you do IMPORTs or RESTOREs, you should do a new full backup after they complete.

I might even say that should just be our primary rec, in the interest of simplicity -- just do a new full backup after any import, restore or create index completes.

Reply:
OK, how’s this to replace the first two bullet points:

- If you create or change indexes, or do an `IMPORT` or `RESTORE`, take a full backup before and after the operation; do not rely on incremental backups taken while doing this operation.
- If you're unsure about the above, then only consider full backups to be reliable. Consider incremental backups to be unreliable, if those incremental backups were taken during in-progress index backfills or imports, or if you are unsure.

Reply:
LGTM. I could take or leave the "before and" -- only the "after" is required to mitigate this issue. You might choose to backup before for other reasons, but I don't know if anything about this issue would motivate recommending it.

Reply:
Good, done.